### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by sundayezeilo**
We are requesting a review on this our first project, bubble sort. You can check out our readme section for details of the project. Your hope your feedback will really be of help to us to improve in both this and other projects.

### CODE REVIEW

**Comment by Anna-Myzukina** [comment]
## Hi team ! Very nice project! :wave: 

# Status: [changes required]

You have made great progress in this project. Your README looks great. I'd give you the 👍  right away however there is one concern we have to address and another for us to consider for next Ruby project.

### Linters
- Address `.stickler.yml` review comment
- You are missing the `.stickler.yml` as seen in the  [linter](https://github.com/microverseinc/linters-config/tree/master/ruby) repo.
- Follow all instructions in the README of the link above to activate your linter.




### Make changes and submit another code review request.

*If you have any questions, let me know on slack (@Anna-Muzykina) and I will be glad to clarify with you.*
*Happy coding * <g-emoji class="g-emoji" alias="computer" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f4bb.png">computer</g-emoji>

**Comment by ezeilo-su** [comment]
TSE's requested changes effected.

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+def bubble_sort(arr)
+  yield(arr) if block_given?
+  arr
+end
```
Comment (majovanilla) [inline bubble_sort.rb:4]: If you are yielding, then your method is not doing anything. This is more likely to be reserved for the bubble_sort_by with some modifications.

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    n = 0
+    until n == length_to_sort
+      if arr[n] > arr[n + 1]
+        arr[n], arr[n + 1] = arr[n + 1], arr[n]
+        is_sorted = false
+      end
+      n += 1
+    end
+    length_to_sort -= 1
+  end
```
Comment (majovanilla) [inline bubble_sort.rb:22]: you should do all this inside the method above, not inside the variable. Please rearrange your code.

**Review by majovanilla — CHANGES_REQUESTED** [review]
@ezeilo-su 

## You did a *nice* job on this project! 💜 

However, there are some more changes you need to implement before moving on. Sorry that the previous TSE didn't mention this, but I'm sure you want to improve your project :smiley: 

- I can't see your `sort_by` method!! That would be the most important thing to work on.
- Your bubble_sort method is not returning the array in increasing order (I get it ordered but backward, from larger to smaller).
- Check the commented code below.

## Please, make all REQUIRED changes and submit your request again! 😀

If you have any questions, feel free to send me a message through Slack (@majo)
I would be happy to help! 👩‍💻 


Happy coding 🤓

--Maria Reyes

**Review by mosesogwo — APPROVED** [review]
**Congratulations. Your project is approved:trophy:**

You can now merge to master

**Cheers:stars:**

### Status: Approved:heavy_check_mark:

**Pull request merged**